src/ds5/ds5-auto-calibration.cpp

```diff
@@ -294,84 +321,166 @@ namespace librealsense
             try_fetch(jsn, "fl scan location", &fl_scan_location);
             try_fetch(jsn, "fy scan direction", &fy_scan_direction);
             try_fetch(jsn, "white wall mode", &white_wall_mode);
+
+            try_fetch(jsn, "scan only", &scan_only_v3);
+            try_fetch(jsn, "interactive scan", &interactive_scan_v3);
+
+            int val = 0;
+            try_fetch(jsn, "step count v3", &val);
+            step_count_v3 = static_cast<uint16_t>(val);
+            if (step_count_v3 > 0)
+            {
+                for (int i = 0; i < step_count_v3; ++i)
+                {
+                    val = 0;
+                    std::stringstream ss;
+                    ss << "fill factor " << i;
+                    try_fetch(jsn, ss.str(), &val);
+                    fill_factor[i] = static_cast<uint16_t>(val);
+                }
+            }
         }
 
         std::vector<uint8_t> res;
+
+        if (host_assistance && _occ_state == RS2_OCC_STATE_NOT_ACTIVE)
+        {
+            _json = json;
+            _action = RS2_OCC_ACTION_ON_CHIP_CALIB;
+            _occ_state = RS2_OCC_STATE_WAIT_TO_CAMERA_START;
+            switch (speed)
+            {
+            case 0:
```

Comment:
There should be enum that defines the scan speed. Pls refactor with that instead of plain numbers

Reply:
Done.